File: 0997. Find the Town Judge/Solution.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        sVoted = set()
        dVotes = dict()
        
        if trust == []:
            return 1 if N == 1 else -1
        
        for t in trust:
            sVoted.add(t[0])
            if t[1] not in dVotes:
                dVotes[t[1]] = 1
            else:
                dVotes[t[1]] += 1
        
        for i in range(1, N+1):
            try:
                if (i not in sVoted) and dVotes[i] == N-1:
                    return i
            except:
                pass
        
        return -1

File: 0997. Find the Town Judge/Solution.cpp
class Solution {
public:
    int findJudge(int N, vector<vector<int>>& trust) {
        std::set<int> sVoted;
        std::map<int, int> mVotes;
        
        for (auto &t : trust)
        {
            sVoted.insert(t[0]);
            mVotes[t[1]] += 1;
        }
        
        for (int i=1;i<N+1;i++)
        {
            auto search = sVoted.find(i);
            if (search == sVoted.end() && mVotes[i] == N-1)
            {
                return i;
            }
        }
        
        return -1;
    }
};
